libwayshot/src/ext_image_protocols.rs

```diff
@@ -0,0 +1,731 @@
+                buffer,
+                physical_size,
+                transform,
+                ..
+            } = data;
+            let surface = compositor.create_surface(&qh, ());
+            let layer_surface = layer_shell.get_layer_surface(
+                &surface,
+                Some(output),
+                Layer::Top,
+                "wayshot".to_string(),
+                &qh,
+                output.clone(),
+            );
+
+            // Configure the toplevel to be fullscreen on the specific output
+            layer_surface.set_exclusive_zone(-1);
+            layer_surface.set_anchor(Anchor::all());
+            layer_surface.set_margin(0, 0, 0, 0);
+
+            debug!("Committing surface creation changes.");
+            surface.commit();
+
+            debug!("Waiting for layer surface to be configured.");
+            while !state.configured_outputs.contains(output) {
+                event_queue.blocking_dispatch(&mut state)?;
+            }
+
+            surface.set_buffer_transform(*transform);
+            // surface.set_buffer_scale(output_info.scale());
+            surface.attach(Some(buffer), 0, 0);
+
+            let viewport = viewporter.get_viewport(&surface, &qh, ());
+            viewport.set_destination(physical_size.width as i32, physical_size.height as i32);
+
+            debug!("Committing surface with attached buffer.");
+            surface.commit();
+            layer_shell_surfaces.push((surface, layer_surface));
+            event_queue.blocking_dispatch(&mut state)?;
+        }
+
+        let region_re = callback.screenshot(self);
+
+        debug!("Unmapping and destroying layer shell surfaces.");
+        for (surface, layer_shell_surface) in layer_shell_surfaces.iter() {
+            surface.attach(None, 0, 0);
+            surface.commit(); //unmap surface by committing a null buffer
+            layer_shell_surface.destroy();
+        }
+        event_queue.roundtrip(&mut state)?;
+        let region = region_re?;
+
+        let shotdata = data_list
+            .iter()
+            .find(|data| data.in_this_screen(region))
+            .ok_or(crate::WayshotError::CaptureFailed(
+                "not in region".to_owned(),
+            ))?;
+        let area = shotdata.clip_area(region).expect("should have");
+        // Use mmap from CaptureOutputData
+        let shotdata_ref = &shotdata.data;
+        let frame_mmap = shotdata_ref.mmap.as_ref().unwrap();
+        let converter = crate::convert::create_converter(shotdata_ref.frame_info.format).unwrap();
+        let mut mmap_vec = frame_mmap.to_vec();
+        let color_type = converter.convert_inplace(&mut mmap_vec);
+        // Return tuple instead of ImageViewInfo
+        Ok((
+            mmap_vec,
+            shotdata_ref.logical_region.inner.size.width,
+            shotdata_ref.logical_region.inner.size.height,
+            color_type,
+            area,
+        ))
+    }
+
+    /// Capture a single output
```

Comment:
this is toplevel capture, so the comment should not be 'capture a single output'